server/parsers/basic_input_output_parser.ts

```diff
@@ -13,19 +13,79 @@ const sanitize = (content: string) => {
   return DOMPurify.sanitize(content, { FORBID_TAGS: ['img'] }).replace(/!+\[/g, '[');
 };
 
+const isEveryItemString = (array?: unknown) =>
+  !!(Array.isArray(array) && array.every((item) => typeof item === 'string'));
+
+export const parseSuggestedActions = (string: string): string[] => {
+  if (!string) {
+    return [];
+  }
+  const suggestedOutputString = string;
+  let suggestedActions: string[] = [];
+  try {
+    suggestedActions = JSON.parse(suggestedOutputString || '[]');
+  } catch (e) {
+    suggestedActions = [];
+  }
+
+  if (suggestedActions.length) {
+    return suggestedActions;
+  }
+
+  /**
+   * Get json-like substring from a string
+   *
+   *  /\{               // Match an opening curly brace
+   *    (?:            // Start a non-capturing group
+   *      [^{}]        // Match any character that is not a curly brace
+   *      |            // OR
+   *      (?:          // Start a non-capturing group
+   *        \{         // Match an opening curly brace
+   *        [^{}]*     // Match any characters that are not curly braces (allowing nested structures)
+   *        \}         // Match a closing curly brace
+   *      )
+   *    )*             // Repeat the non-capturing group zero or more times (allowing nested structures)
+   *  \}/g             // Match a closing curly brace, and 'g' flag for global search
+   *
+   */
+  const jsonPattern = /\{(?:[^{}]|(?:\{[^{}]*\}))*\}/g;
```

Comment:
How about add response key to the jsonPattern? We only want to use this key in the final steps.

Reply:
You mean only parse suggested actions in latest interaction right? Good suggestion and will do that.

Reply:
I mean we can match with the response key like `'{"response":'`. It will be more accurate.

Reply:
Agree, and add a check to only parse suggested actions for the last interaction.